crates/bevy_utils/macros/src/lib.rs

```diff
@@ -35,12 +35,55 @@ impl Parse for AllTuples {
     }
 }
 
+/// Helper macro to generate tuple pyramids. Useful to generate scaffolding to work around Rust
+/// lacking variadics. Invoking `all_tuples!(impl_foo, start, end, P, Q, ..)`
+/// invokes `impl_foo` providing ident tuples through arity `start..=end`.
+/// ```
+/// use bevy_utils_proc_macros::all_tuples;
+///
+/// trait Foo {
+///     // ..
+/// }
+///
+/// macro_rules! impl_foo {
+///     ($($P:ident),*) => {
+///         // ..
```

Comment:
```suggestion
///         impl<$($P),*> Foo for ($($P,)*) {
///             // ..
///         }
```

This would be more helpful if it showed what an impl is supposed to look like. Also, this lets the example double as a doctest somewhat.